toml/parser.ts

```diff
@@ -0,0 +1,312 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+import { existsSync } from "../fs/exists.ts";
+
+const deepAssign = (target: object, ...sources: object[]): object => {
+  for (let i = 0; i < sources.length; i++) {
+    const source = sources[i];
+    if (!source || typeof source !== `object`) {
+      return;
+    }
+    Object.entries(source).forEach(([key, value]) => {
+      if (value instanceof Date) {
+        target[key] = new Date(value);
+        return;
+      }
+      if (!value || typeof value !== `object`) {
+        target[key] = value;
+        return;
+      }
+      if (Array.isArray(value)) {
+        target[key] = [];
+      }
+      // value is an Object
+      if (typeof target[key] !== `object` || !target[key]) {
+        target[key] = {};
+      }
+      deepAssign(target[key], value);
+    });
+  }
+  return target;
+};
+
+const regGroup = /\[.*\]/;
+class KeyValuePair {
+  key: string;
+  value: unknown;
+}
+class Parser {
+  tomlLines: string[];
+  context: any;
```

Comment:
Can this be given a non-any type? (It seems lint shouldn't allow any)

Reply:
Have to do a bit of refactor to make it work but it's in the scope yep.

Reply:
corrected